src/libopensc/cwa14890.c

```diff
@@ -1896,6 +1883,9 @@ int cwa_decode_response(sc_card_t * card,
 	}
 
 	/* finally apply 3DES to result */
+	sc_evp_cipher_free(alg);
+	alg = sc_evp_cipher(card->ctx, "DES-EDE-ECB");
+
 	if (EVP_EncryptInit_ex(cctx, EVP_des_ede_ecb(), NULL, key, NULL) != 1 ||
```

Comment:
this is missing the `alg` usage as defined on the previous line:
```suggestion
	if (EVP_EncryptInit_ex(cctx, alg, NULL, key, NULL) != 1 ||
```

Reply:
good catch